CRM/Contribute/PseudoConstant.php

```diff
@@ -379,36 +372,25 @@ public static function &pcpStatus($column = 'label') {
   }
 
   /**
-   * Get all financial accounts for a Financial type.
+   * Get financial account for a Financial type.
    *
-   * The static array  $financialTypeAccount is returned
    *
+   * @param int $entityId
+   * @param string $accountRelationType
    *
-   * @param int $financialTypeId
-   * @param int $relationTypeId
-   * @return array
-   *   array reference of all financial accounts for a Financial type
+   * @return int
    */
-  public static function financialAccountType($financialTypeId, $relationTypeId = NULL) {
-    if (!CRM_Utils_Array::value($financialTypeId, self::$financialTypeAccount)) {
-      $condition = " entity_id = $financialTypeId ";
-      CRM_Core_PseudoConstant::populate(
-        self::$financialTypeAccount[$financialTypeId],
-        'CRM_Financial_DAO_EntityFinancialAccount',
-        $all = TRUE,
-        $retrieve = 'financial_account_id',
-        $filter = NULL,
-        $condition,
-        NULL,
-        'account_relationship'
-      );
+  public static function getRelationalFinancialAccount($entityId, $accountRelationType) {
+    $result = civicrm_api3('EntityFinancialAccount', 'get', array(
+      'return' => array("financial_account_id"),
+      'account_relationship.name' => $accountRelationType,
+      'entity_table' => 'civicrm_financial_type',
+      'entity_id' => $entityId,
+    ));
```

Comment:
@eileenmcnaughton I have removed the 3rd parameter that was representing ```entity_table```. However, using ```Civi::$statics``` won't be applicable here because of dynamic factor ```$entityId``` currently looking at the examples at ```CRM_Core_Pseudoconstant``` most of them support level 2 combination . But in this case it